xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -392,7 +391,7 @@ func (b *cdsBalancer) run() {
 				b.edsLB = nil
 			}
 			b.xdsClient.Close()
-			// This is the *ONLY* point of return from this function.
+			b.done.Fire()
```

Comment:
It's a nit, but I'd move this line after logging so we don't use anything in `b` at all after firing `done`.

Actually, should we log before doing the cancelWatch, even, so we know when shutdown was _initiated_?  I'm okay either way, but that seems potentially more useful.

Reply:
I moved the log line to before firing `done`. 

As far as moving it to before `cancelWatch` and using it to indicate initiation of the shutdown process, I feel that the `Shutdown` log line more corresponds to actual and complete shutdown rather than initiation of it. So, I will leave it here. If we think it is better to use it to indicate initiation of the shutdown process, we can do that, but we will have to change in a lot of other places as well.